aggregator.py

```diff
@@ -0,0 +1,183 @@
+import logging
+import Queue
+import time
+
+from Core.message_object import BeliefMessage
+from Core.process import Process
+
+from belief_manager import BeliefManager
+
+
+logger = logging.getLogger(__name__)
+
+
+def _moving_average(sample, count, average):
```

Comment:
I think your way saves exactly one multiplication, which *might* be worth thinking about if we were writing, say, an interrupt handler routine for a microcontroller that had to run thousands of times a second...